providers/summary/scancode.js

```diff
@@ -178,6 +189,10 @@ class ScanCodeSummarizer {
       .filter(e => e)
   }
 
+  _isMatchedLicense(license) {
+    return license.score >= 80
```

Comment:
where is this threshold documented/defined for the community understanding - is that a scancode native value that means "definitely the match"?

Reply:
[I read the more detailed commit messages]

Are there any situations where this >= 80 threshold might produce an inaccurate resulting license that we could point to a risk on this?

Reply:
Thanks for the feedback!
The exiting logic in scancode summarizer includes licenses with lower scores in the declared license, but lists only licenses scoring above 80 for the files.  For example, the declared licenses for https://clearlydefined.io/definitions/composer/packagist/colinmollenhour/cache-backend-redis/1.14.4 are BSD-3-Clause AND GPL-2.0-only.  However discovered license is only BSD-3-Clause (GPL-2.0-only is filtered out due to lower score).  This inconsistency causes confusion.  My attempt here is to make declared license and discovered license consistent and remove licenses scoring less than 80 from the declared license.

As to what is the appropriate score threshold to avoid risk of incorrect license:
- in versions prior to 3.0.0, licenses with score >= 90 are considered as declared license (`_getLicenseByFileName`)
- Since version 3.0.0, is_license_text flag from scancode is used.  The “is_license_text” flag is set to true for files that contain mostly license texts and notices.  Based on [documentation](https://github.com/nexB/scancode-toolkit/blob/6d4ea7f10cf0d92f1ce922ff46e85e2e903ff4e8/docs/source/cli-reference/scan-options-post.rst#--is-license-text-option_), "mostly means over 90% of the content of the file."
According to this documentation (prior to v32 update, 30.1.0 is currently used in the crawler), the license objects for each detected license in the files has "is_license_text" attribute as well.  I will adapt my solution to use the is_license_text flag on the detected license instead.

Reply:
Upon further research, there are four flags determined at each license match level: "is_license_text", "is_license_notice", "is_license_tag", "is_license_reference".   This is also the order of precedence.  Details see https://scancode-results-analyzer-fork-ayan.readthedocs.io/en/latest/how-analysis-is-performed/cases-incorrect-scans.html.  is_license_text alone is not enough to determine whether it is the correct license match.  Also see more discussion at  https://github.com/nexB/scancode-toolkit/issues/3082

I will refactor and reuse the logic for scancode prior to 3.0.0 to determine declared license by scoring >= 90.  This way is more consistent within CD's implementation.  @jeffwilcox What do you think?

Reply:
> Are there any situations where this >= 80 threshold might produce an inaccurate resulting license that we could point to a risk on this?

@jeffwilcox Without any fixes, there is currently a risk of inaccurate resulting license.  Details see [issue](https://github.com/clearlydefined/service/issues/1010).  File licenses (as discovered) are always filtered based on score (>= 80).  For declared licenses, there is a filtering based on score (>=90) in place for ScanCode data with versions prior to 3.0.0.  This filtering is missing for later versioned ScanCode data, resulting license matched with lower scores shows up in "declared", but filtered out in "discovered".  Including the score filtering (>=90) will ensure more accurate license information, and also consistency in handling different versions of ScanCode data.